### PR TITLE
Initial 510 world.tick_usage integration

### PR DESCRIPTION
**Pull request by ghost**
Yep, for real.

### CODE REVIEW

File: core/process.dm [hunk 1]
```diff
@@ -197,19 +199,15 @@ datum/controller/process/proc/scheck(var/tickId = 0)
 
 	// For each tick the process defers, it increments the cpu_defer_count so we don't
 	// defer indefinitely
-	if (main.getCurrentTickElapsedTime() > main.timeAllowance)
-		sleep(world.tick_lag*1)
+	if (world.tick_usage > 100 || main.world.tick_usage > tick_start + tick_allowance)
```
Comment (ghost) [inline core/process.dm]: wow what the fuck is this line
not sure how that snuck in there

**Comment by Fox-McCloud** [comment]
@docsingh 

You may be interested in some of the work one of Paradise's coders (@tastyfish) has done on this---it completely eliminates dependency on btime, to boot (and referencing it is expensive as hell--just profile scheck after a 1-2 hour round).

https://github.com/ParadiseSS13/Paradise/pull/3747

**Comment by volundr-** [comment]
@Fox-McCloud Yep, this is neat. Conceptualizing TimeOfTick and TimeOfGame as a recalculation of world.tick_usage is a nice way to handle timing in the game.

Eliminating btime is one of @docsingh's goals here. Btime is annoying to set up, expensive to use, and a horrible hack that should never have had to exist in the first place. Now that we have a better timing source, we don't need btime.

One thing I notice in @tastyfish's changes is that he didn't eliminate the updateCurrentTickData stuff - That is now cruft and can be removed, as it was only intended to provide a reasonable estimate of how much time had elapsed in the current tick. With btime, it was able to do so with something like a 10% margin of error. Without btime, the margin of error was as much as 50%. Since we now have world.tick_usage, we don't need to try and estimate this anymore.

**Comment by Tastyfish** [comment]
The reason why I kept that stuff is that tg-derived servers have a process status panel, which uses those extra vars for profiling. Looks like goon could completely cut it out, however, as they don't do that.

**Comment by volundr-** [comment]
@tastyfish that stuff is pretty resource intensive. Best to jettison it as soon as you can.

**Pull request merged**